ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/Flow.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package io.confluent.ksql.rest.server.resources.streaming;
+
+import org.apache.kafka.connect.data.Schema;
+
+public class Flow {
+
+  public interface Subscriber<T> {
+
+    void onNext(T item);
+
+    void onError(Throwable e);
+
+    void onComplete();
```

Comment:
yes, hopefully we'll have a sane way of indicating the end of the stream for limit queries at some point, instead of letting and unhandled exception bubble up from kafka streams.

Reply:
I guess i'd add it when it is needed. Not a fan of having unused methods etc

Reply:
@dguy looks like this is getting addressed in https://github.com/confluentinc/ksql/pull/1337, so once that's merged I can update this PR to use onComplete